tools/build/build.sh

```diff
@@ -50,7 +50,9 @@ if (( $# > 0 )); then
   readonly GIT_ROOT=$(git rev-parse --show-toplevel)
   # Rather than a working directory of something like "/worker", mirror
   # the path on the host so that symlink tricks work as expected.
-  docker run --rm -ti -v "$GIT_ROOT":"$GIT_ROOT" -w "$GIT_ROOT" $IMAGE_NAME "$@"
+  # We cache ~/.gradle between builds, for speed.  We also ensure the parent folder exists.
+  mkdir -p "$GIT_ROOT/build/.gradle-docker"
+  docker run --rm -ti -v "$GIT_ROOT":"$GIT_ROOT" -w "$GIT_ROOT" -v "$GIT_ROOT/build/.gradle-docker:/root/.gradle" $IMAGE_NAME "$@"
```

Comment:
Just notes: When moving from `/tmp` to `build` I had to address two concerns:

- If the `build` folder does not exist, I fear docker will fail to mount `build/.gradle-docker`, so I added a `mkdir -p`

- There is a danger than `$GIT_ROOT` may contain spaces, so I had to drop the `CACHE_GRADLE` descriptive variable, and instead put the `-v` option directly inline.